connectors/sources/mysql.py

```diff
@@ -252,7 +329,7 @@ async def validate_config(self):
         strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
     )
     async def _remote_validation(self):
-        async with self.with_connection_pool() as connection_pool:
+        async with self._mysql_client.with_connection_pool() as connection_pool:
```

Comment:
This will be replaced in the future with a more high-level method. `_with_connection_pool` will probably be an internal method of `MySQLClient`. Just wanted to keep the PR small as possible for now.

Reply:
yeah, I don't think we need to surface two layers -- the code just needs a pool, the rest is internal implementation detail. I guess in the future that might change if one connector talks to 2 different SQL servers